### ISSUE TITLE
Correct `pyproject.toml` build-backend specification

### ISSUE THREAD

**Issue by nchepanov** [graphql-python/graphql-core#118]
Currently:

https://github.com/graphql-python/graphql-core/blob/7d826f0ec0a447cb869dfb891a755dbe8ea9b66f/pyproject.toml#L54-L56

According to https://python-poetry.org/docs/pyproject/#poetry-and-pep-517

> If your `pyproject.toml` file still references poetry directly as a build backend, you should update it to reference poetry_core instead.

Should be updated  to,

```toml
[build-system]
requires = ["poetry_core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

This will significantly speed up the package installation process, since the entire `poetry` dependency cone is currently being installed during isolated build per PEP517.

I can submit a PR if no objections.

**Comment by KingDarBoja** [comment]
Good catch, feel free to submit a PR 🚀

**Comment by Cito** [comment]
Thanks for reporting. This was fixed in add666f36116e1bde1eba3d54e7ac8ecc64a5420 and will be available in the upcoming release (soon).

**Comment by Cito** [comment]
[Version 3.1.3](https://github.com/graphql-python/graphql-core/releases/tag/v3.1.3) with this fix has now been released.
